Boltdir/site-modules/gplt/tasks/create_control_repo_from_production_env.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+ENVIRONMENT=production
+PUPPET_PROD_ENV_DIR=/etc/puppetlabs/code/environments/$ENVIRONMENT
+BASEDIR=/opt/puppet
+CONTROL_REPO_URL=$BASEDIR/control-repo.git
+CHECKOUT_DIR=/root/control-repo
+
+mkdir -p $BASEDIR
+
+if [ -d $CONTROL_REPO_URL ]; then
+    echo "ERROR: The control repo destination already exists" >&2
+    exit 1
+fi
+
+git init --bare $CONTROL_REPO_URL
+git clone $CONTROL_REPO_URL $CHECKOUT_DIR
+cp -r $PUPPET_PROD_ENV_DIR/* $CHECKOUT_DIR
+rm -fr $CHECKOUT_DIR/modules
+cd $CHECKOUT_DIR \
+    || { echo "ERROR: Failed to cd into $CHECKOUT_DIR" >&2 ; exit 1 ; }
+git add .
+git commit -m "Initial commit"
+git branch -m $ENVIRONMENT
+git push -u origin $ENVIRONMENT
```

Comment:
When I tried this general flow, then tried to clone the repo to a new place I got this 
warning: remote HEAD refers to nonexistent ref, unable to checkout.
I think we need to change the default branch to be production.

Reply:
Line 24 moves the default `master` branch to `production`. This line then pushes the production branch to the remote.  Since the remote was bare to begin with, the only branch that exists after this line is the `production` branch.

Reply:
yeah, but git still throws an error if you clone it.  And the branch is set to master sort of...

➜  blue git clone /tmp/foo.git
Cloning into 'foo'...
done.
warning: remote HEAD refers to nonexistent ref, unable to checkout.

➜  blue cd foo
➜  foo git:(master)

Though git branch returns nothing.  And if you checkout production can you can't checkout master to go back.  But you end up in a really odd state...  And in fact you don't have any files present after the initial clone.

This is what determines the default branch
➜  foo git:(master) cat .git/HEAD
ref: refs/heads/master

running
git symbolic-ref HEAD refs/heads/production
after cd'ing into $CONTROL_REPO_URL
even before you created the production branch will fix it.

Reply:
Thanks for finding this @RandellP . I have amended the script to update the bare repo as you have indicated.